uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -1594,6 +1594,69 @@ public function testGetShouldOnlyReturnTheLeaveRequestsOfStaffMembersManagedByMa
     $this->assertEquals(0, $result['count']);
   }
 
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: leave_request_id
+   */
+  public function testIsManagedByShouldThrowAnExceptionIfLeaveRequestIDIsMissing() {
+    civicrm_api3('LeaveRequest', 'isManagedBy', ['contact_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: contact_id
+   */
+  public function testIsManagedByShouldThrowAnExceptionIfContactIDIsMissing() {
+    civicrm_api3('LeaveRequest', 'isManagedBy', ['leave_request_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: leave_request_id, contact_id
+   */
+  public function testIsManagedByShouldThrowAnExceptionIfBothTheContactIDAndLeaveRequestIDAreMissing() {
+    civicrm_api3('LeaveRequest', 'isManagedBy');
+  }
+
+  public function testIsManagedByShouldReturnTrueIfTheContactOfTheGivenLeaveRequestIsManagedByTheGivenContactID() {
```

Comment:
This first assertion is only to make sure the relationship doesn't exist prior the call to `setContactAsLeaveApproverOf()`. You can see I use the same approach for the other tests on LeaveManager. It's just a way to be sure the method would not return true even if the relationship between the two contacts doesn't exist.

Ideally this should be 2 tests, but we're starting to have performance problems with them (currently, it takes 1.35 minutes to run all of them on my computer), so I prefer to avoid splitting simple tests like this one. Of course, if this was something more complex with lots of changes between the first and last assertion, then I would definitely split it.

I don't see the contraction on the test name. It says it should return true if the contact of the leave request is managed by the given contact id, and that is what happens, the assertions make sure true is returned only on the describe scenario. What is the contradiction?